### PR TITLE
Cover block: improve overlay opacity handling

### LINKED ISSUE #25290: custom Cover Block overlay color overwritten by "has-background-dim"
**Describe the bug**
When adding a cover block and picking a color registered by a theme/plugin, the "has-background-dim" class overrides the custom class styles that's added by the theme/plugin.

Here's the CSS classes constructed for the cover block:

class="wp-block-cover has-wpbf-palette-color-2-background-color has-background-dim"

The problem is that the "has-background-dim" class is added AFTER the "has-wpbf-palette-color..." class which makes it override the color.

Because the color tweak is the "more granular" change, the "has-wpbf-palette-color..." class should be added at the end, after the has-background-dim class.

**To reproduce**
1. Create a color block
2. Choose a custom overlay color that was previously registered by a theme
3. Update the post/page & check the result on the frontend.
4. Notice that "has-background-dim" has a higher priority as the custom color added by the theme

**Expected behavior**
The "has-wpbf-palette-color..." class should be added after "has-background-dim".

**Editor version (please complete the following information):**
- WordPress version: 5.5.1
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? Default

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser Firefox

### PR DESCRIPTION
**Pull request by stokesman**
Fixes: #25290 which I believe was created by the change from #20904 as it moved the assignment of the black background color into a selector with greater specificity. That PR and its comments are also relevant here because in them an alternative fix was suggested by @jasmussen which is part of the changes in this PR. That is removing the styled fallback background color from the cover. (I'm not referring to it as the default background color because it's just styled and not part of the block attributes). The idea was discarded since it would "break" the dimming feature but if there's no background color or gradient set, that feature just shouldn't be available. Which is what has been done in this PR.

Besides that this gives the opacity range slider a `step` in accord with the available opacity styles 68419b4 and includes a very minor CSS optimization 0fb0fed. 

Note: I'd say part of the issue in #25290 is that the `has-background-dim` class is inappropriately applied to the block. I initially started on a fix by improving the logic for when that class is applied. However, that leads to block validation failures. If there's a way to handle the migration/upgrade of a block then that'd still be a nice thing to fix.

<details>
<summary>Screen recording demonstrating availability of opacity control only with application of background color (also stepped opacity increments)</summary>

Also worth noting that the background image used in that recording would not even be visible without the changes in this PR.
</details>

## How has this been tested?
WordPress 5.5.1, creating various Cover blocks with and without images and background colors.

## Types of changes
Bugfix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by jasmussen** [comment]
Very nice. This is my before:

before

And this is after:

after

And so as you can see, purely based on those GIFs, this PR solves that issue. 

I also verified that this fixes the issue for the _overlay_ aspect as well. Editor:

Screenshot 2020-10-15 at 10 41 42

Frontend:

Screenshot 2020-10-15 at 10 41 50

Currently all overlays are black 🙈 so this would be a nice PR to land!

**Review by mkaz — APPROVED** [review]
This looks good, tested well. No problems 👍

File: packages/block-library/src/cover/style.scss [hunk 1]
```diff
@@ -34,13 +34,9 @@
...
-	&.has-background-dim {
-		background-color: $black;
-
-		&::before {
-			content: "";
-			background-color: inherit;
-		}
+	&.has-background-dim::before {
+		content: "";
+		background-color: inherit;
```
Comment (sirreal) [inline packages/block-library/src/cover/style.scss:39]: Should this have become `$black` because the inherited background-color was removed?

**Review by sirreal — COMMENTED** [review]
This may have introduced a regression, I've just filed an issue: https://github.com/WordPress/gutenberg/issues/26545

File: packages/block-library/src/cover/style.scss (hunk 1 above)
Comment (stokesman) [reply packages/block-library/src/cover/style.scss:39]: If that were to be `$black` themes would have to write some quite different styles to have their colors apply on Covers. The removal was motivated by pretty much that issue #25290, the way it made theme background style selectors need extra specificity. As an example, from twentytwenty:
`:root .has-primary-background-color { ... }`
Whereas the [docs on defining those](https://developer.wordpress.org/block-editor/developers/themes/theme-support/#block-color-palettes) use a simple class selector.
 Also, the `inherit` here is for when there's a defined background color via the block attributes.

Comment (jasmussen) [reply packages/block-library/src/cover/style.scss:39]: I have a "fix" in https://github.com/WordPress/gutenberg/pull/26564, that I'd love your thoughts on.

**Pull request merged**